### DISCUSSION TITLE
如何修改algorithm2e包的caption序号格式

### DISCUSSION THREAD

**Discussion by MARMOTatZJU** [TheNetAdmin/zjuthesis#114] [General]
在`config/format/general/reference.tex`可见一些figure/table/lstings到中文的映射，想请教对于基于alrogithm2e包写的算法，如何修改才能使得下面显示的caption中algorithm显示为中文“算法”二字？

![image](https://user-images.githubusercontent.com/32362315/103399301-63a8ce80-4b7b-11eb-9bc4-7a065d67c437.png)

**Comment by TheNetAdmin** [comment]
大概这样可以解决问题

```TeX
\def\algorithmcfname            {算法}
\def\algorithmautorefname       {算法}
```

![2020-12-31_00-05](https://user-images.githubusercontent.com/18525442/103401002-091c5d80-4afc-11eb-8d4d-ed920950fd3c.png)

**Reply by TheNetAdmin** [reply]
代码已更新

**Reply by MARMOTatZJU** [reply]
这个可以用，感谢！
BTW，请问这两行命令的名字是在哪里找到的？有系统的介绍autorefname的manual吗？

**Reply by muzimuzhi** [reply]
> 大概这样可以解决问题
> 
> ```
> \def\algorithmcfname            {算法}
> \def\algorithmautorefname       {算法}
> ```
> 

`algorithm2e` 没有暴露这两个命令，宏包文档只在 sec. 6 提到了 `\SetAlgorithmName{name}{autoref name}{list name}`。它的实现 ([`algorithm2e.sty` v5.2, lines 2881--2885](http://www.tug.org/svn/texlive/trunk/Master/texmf-dist/tex/latex/algorithm2e/algorithm2e.sty?view=markup&pathrev=44846#l2881))
```tex
\newcommand{\SetAlgorithmName}[3]{%
  \renewcommand{\listalgorithmcfname}{#3}%
  \renewcommand{\algorithmcfname}{#1}%
  \renewcommand{\algorithmautorefname}{#2}%
}%
```
提醒了，必要时也要重定义 `\listalgorithmcfname`。

当前模板不依赖 `algorithm2e`，是用户选择了这个伪代码包。感觉未必需要提供它的配置。

**Reply by TheNetAdmin** [reply]
> 提醒了，必要时也要重定义 `\listalgorithmcfname`。

感谢提醒，我已经将 `\listalgorithmcfname` 加入代码
 
> 当前模板不依赖 `algorithm2e`，是用户选择了这个伪代码包。感觉未必需要提供它的配置。

我个人没有在模板中用过 algorithm 各种包，所以没有汉化过相关条目

但如果这里的三个设置可以在各种 algorithm 相关的包通用的话，感觉可以保留以方便其他使用算法包的同学

**Reply by MARMOTatZJU** [reply]
> 当前模板不依赖 `algorithm2e`，是用户选择了这个伪代码包。感觉未必需要提供它的配置。

@muzimuzhi 目前论文里有描述流程/控制流的需求，平时看到的相关领域的会议论文里也大多有用伪代码来描述一些流程/控制流，所以“用伪代码描述相对复杂的流程/控制流”应该是一个不小的需求。

所以请问比较主流的描述算法的包应该是？algorithm2e不是很主流吗（这个是从别人放arxiv上的会议论文的latex源码里看到的）？

**Reply by muzimuzhi** [reply]
> 但如果这里的三个设置可以在各种 algorithm 相关的包通用的话，感觉可以保留以方便其他使用算法包的同学

不通用。
